usm_rest_api/v1/views/tasks.py

```diff
@@ -0,0 +1,224 @@
+        return {'status': 'ALL_NODES_FAILURE', 'failednodes': str(failedNodes)}
+    elif failedNodes:
+        log.critical("Cluster creation partially Failed %s and Nodelist %s" %
+                     (cluster_data, failedNodes))
+        return {'status': 'PARTIAL_FAILURE', 'failednodes': str(failedNodes)}
+
+    return states.SUCCESS
+
+
+@shared_task
+def createHost(data):
+    log.debug("Inside createGlusterHost Async Task %s" % data)
+    cluster = None
+    cluster_data = None
+    hostlist = None
+
+    # Get the cluster configuration
+    try:
+        # Get the cluster config
+        cluster = Cluster.objects.get(pk=str(data['cluster']))
+        cluster_data = {"cluster_id": cluster.cluster_id,
+                        "cluster_name": cluster.cluster_name,
+                        "cluster_type": cluster.cluster_type,
+                        "storage_type": cluster.storage_type}
+        log.debug("Cluster Data %s" % cluster_data)
+        # Get nodes belongs to the cluster for peerprobe
+        hostlist = Host.objects.filter(cluster_id=str(data['cluster']))
+        log.debug("Hosts in the cluster: %s" % hostlist)
+    except Exception, e:
+        log.exception(e)
+        return {'status': 'HOST_CREATION_FAILURE', 'error': str(e)}
+
+    # Setup the host communication
+    current_task.update_state(state='ESTABLISH_HOST_COMMUNICATION')
+    minionIds, failedNodes = usm_rest_utils.setup_transport_and_update_db(
+        cluster_data, [data])
+    if failedNodes:
+        log.debug("Failed to Establish Transport with Node %s" % failedNodes)
+        return {
+            'status': 'HOST_CREATION_FAILURE',
+            'error': "Failed to Establish Transport with Node %s"
+            % failedNodes}
+    log.debug("setup_transport_and_update_db done. minionIds %s failedNodes \
+              %s" % (minionIds, failedNodes))
+    # Add the Host to cluster based on the node type
+    # ie if the node type is monitor, then create a monitor
+    # else create the OSD
+    if data['node_type'] == usm_rest_utils.HOST_TYPE_MONITOR:
+        log.debug("Add Monitor Host")
+        current_task.update_state(state='ADD_MONITOR_HOST')
+        failedNodes = salt_wrapper.add_ceph_mon(
+            cluster.cluster_name,
+            {usm_wrapper_utils.resolve_ip_address(data['management_ip']):
+                {'public_ip': data['public_ip']}})
+        if failedNodes:
+            log.debug("Failed to add monitor to cluster %s" % failedNodes)
+            return {
+                'status': 'HOST_CREATION_FAILURE',
+                'error': "Failed to add monitor to cluster %s"
+                % failedNodes}
+        log.debug("After Monitor Host Success %s" % str(failedNodes))
+    elif data['node_type'] == usm_rest_utils.HOST_TYPE_OSD:
+        current_task.update_state(state='ADD_OSD_HOST')
+        # Code to add OSD will go here
+        log.debug("Add OSD Host")
+    elif data['node_type'] == usm_rest_utils.HOST_TYPE_GLUSTER:
```

Comment:
This function sees to deal with both ceph and gluster. I consider that to be unnecessary complexity. I suggest that you break this up in to smaller functions that deal with the specifics